pkg/theiactl/commands/check.go

```diff
@@ -0,0 +1,176 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	sparkv1 "antrea.io/theia/third_party/sparkoperator/v1beta2"
+)
+
+// checkCmd represents the check command
+var checkCmd = &cobra.Command{
+	Use:   "check",
+	Short: "Check the status of a policy recommendation Spark job",
+	Long: `Check the current status of a policy recommendation Spark job by ID.
+It will return the status of this Spark application like SUBMITTED, RUNNING, COMPLETED, or FAILED.`,
+	Example: `
+Check the current status of job with ID e998433e-accb-4888-9fc8-06563f073e86
+$ theiactl policyreco check --id e998433e-accb-4888-9fc8-06563f073e86
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		recoID, err := cmd.Flags().GetString("id")
+		if err != nil {
+			return err
+		}
+		_, err = uuid.Parse(recoID)
+		if err != nil {
+			return fmt.Errorf("failed to decode input id %s into a UUID, err: %v", recoID, err)
+		}
+
+		kubeconfig, err := cmd.Flags().GetString("kubeconfig")
+		if err != nil {
+			return err
+		}
+		clientset, err := CreateK8sClient(kubeconfig)
+		if err != nil {
+			return fmt.Errorf("couldn't create k8s client using given kubeconfig, %v", err)
+		}
+
+		err = PolicyRecoPreCheck(clientset)
+		if err != nil {
+			return err
+		}
+
+		sparkApplication := &sparkv1.SparkApplication{}
+		err = clientset.CoreV1().RESTClient().
+			Get().
+			AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
+			Namespace(flowVisibilityNS).
+			Resource("sparkapplications").
+			Name("policy-reco-" + recoID).
+			Do(context.TODO()).
+			Into(sparkApplication)
+		if err != nil {
+			return err
+		}
+		state := strings.TrimSpace(string(sparkApplication.Status.AppState.State))
+		if state == "RUNNING" {
+			// Check the working progress of running recommendation job
+			// Forward the policy recommendation service port
+			serviceName := fmt.Sprintf("service/policy-reco-%s-ui-svc", recoID)
+			portForwardCmd := exec.Command("kubectl", "port-forward", serviceName, "-n", flowVisibilityNS, "4040:4040")
+			if err := portForwardCmd.Start(); err != nil {
+				return fmt.Errorf("failed to forward port for policy recommendation service, %v", err)
+			}
+			defer portForwardCmd.Process.Kill()
+			stateProgress, err := getPolicyRecommendationProgress("http://localhost:4040")
+			if err != nil {
+				return fmt.Errorf("failed to get the status of job with id %s: %v", recoID, err)
+			}
+			state += stateProgress
+		}
+		fmt.Printf("Status of this policy recommendation job is %s\n", state)
+		return nil
+	},
+}
+
+func getPolicyRecommendationProgress(baseUrl string) (string, error) {
+	// Get the id of current spark application
+	url := fmt.Sprintf("%s/api/v1/applications", baseUrl)
```

Comment:
For a non-existent id (for a typo issue), what state we will show?

Reply:
For a non-existent id, we will fail when getting status from Spark Application CRD instance with some error like 
```
Error: sparkapplications.sparkoperator.k8s.io "<non-existent id>" not found
```